tests/utils/test_batch_helpers.py

```diff
@@ -0,0 +1,200 @@
+from collections import ChainMap, Counter, OrderedDict, defaultdict, deque
+from typing import NamedTuple
+
+import numpy as np
+import pytest
+import torch
+
+from composer.utils.batch_helpers import batch_get, batch_set
+
+my_list = [3, 4, 5, 6, 7, 8, 9, 10]
+
+keys = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
+
+
+class MyClass(object):
+
+    def __init__(self, **kwargs):
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+
+
+list_types = [type(element) for element in my_list]
+my_named_tuple = NamedTuple('nt', **dict(zip(keys, list_types)))
+counter_list = []
+for char, num in zip(keys, my_list):
+    counter_list.extend(num * [char])
+
+
+@pytest.fixture(scope="module",
+                params=[
+                    my_list,
+                    tuple(my_list),
+                    torch.tensor(my_list),
+                    np.asarray(my_list),
+                    deque(my_list),
+                    my_named_tuple(*my_list)
+                ])
+def example_sequence(request):
+    return request.param
+
+
+@pytest.fixture(scope="module",
+                params=[my_list,
+                        tuple(my_list),
+                        torch.tensor(my_list),
+                        np.asarray(my_list),
+                        my_named_tuple(*my_list)])
+def example_deque_less_sequence(request):
+    return request.param
+
+
+# All key value pair data structures that have a __getitem__ function thats takes str.
+@pytest.fixture(scope="module",
+                params=[
+                    dict(zip(keys, my_list)),
+                    MyClass(**dict(zip(keys, my_list))),
+                    my_named_tuple(*my_list),
+                    defaultdict(list, **dict(zip(keys, my_list))),
+                    ChainMap(dict(zip(keys, my_list)), dict(a=7, j=3)),
+                    Counter(counter_list),
+                    OrderedDict(**dict(zip(keys, my_list)))
+                ])
+def example_map(request):
+    return request.param
+
+
```

Comment:
Could you also add tests for where the key is a torch tensor or numpy array? Mainly want to ensure that this will trigger `batch_get_item` rather than `batch_get_sequence`.

Reply:
I'm not sure what you mean this?

Reply:
Something like

```python
x = torch.tensor([[1, 2], [3, 4]])
y = torch.tensor([[True, False], [True, False]])
z = torch.tensor([1, 3])
assert z == batch_get(batch=x, key=y)
```

Reply:
ok added the functionality for keys of type lists, tuples, arrays, tensors, where the batch is of type array or tensor here (with tests for tuple and list keys): [4a1373f](https://github.com/mosaicml/composer/pull/978/commits/4a1373f07a36bc1f23ceb7b28093b119c45c1da2) and then added tests for keys of type tensor or array here: [937ea33](https://github.com/mosaicml/composer/pull/978/commits/937ea33f4313d717cf1fcaac8879f32b089bb6e7)